### PR TITLE
fix!: Combine exporter and endpoint

### PR DESCRIPTION
**Pull request by gbbirkisson**
Since this https://github.com/kolonialno/troncos/pull/282 introduces a breaking change to the api, I wanted to use the opportunity to merge endpoint and exporter into one param in `configure_tracer`.

### CODE REVIEW

**Comment by andersliland** [comment]
A release (without a major update has been published)...

https://github.com/kolonialno/troncos/releases/tag/v4.4.0

Adding release-please to prevent this happening again...

@eirsyl

**Comment by eirsyl** [comment]
Yeah I know. That's whats happens if we just merge stuff to main and let it live there without a release for a longer period.

**Comment by andersliland** [comment]
Yes was not a good move. Release please will fix this.

**Comment by gbbirkisson** [comment]
@eirsyl Still waiting for a review on this PR ...

**Comment by eirsyl** [comment]
Yeah, I can review it. It is better to ask directly if you are waiting on me. Just assigning PRs does not work anymore. 

Skjermbilde 2024-02-05 kl  09 51 51

This is just half the list...

File: README.md [hunk 1]
```diff
@@ -87,7 +87,10 @@ TRACE_PORT = "4318"
 configure_tracer(
     enabled=False, # Set to True when TRACE_HOST is configured.
     service_name='SERVICE_NAME',
-    endpoint=f"http://{TRACE_HOST}:{TRACE_PORT}/v1/traces",
+    exporter=Exporter(
```
Comment (eirsyl) [inline README.md:90]: Can we simplify this a bit further when we first look into this? 

```suggestion
    exporter=Exporter(
      host=TRACE_HOST,
      port=TRACE_PORT,
      exporter_type=ExporterType.HTTP,
      scheme: Optional[str] = None,
      path: Optional[str] = None,
    )
```

We can probably just use `http` as the default scheme, and add `/v1/traces` as the path when http is used. We can probably do this to make it easier for people to use the library.

File: troncos/tracing/__init__.py [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
+_env_port = os.environ.get("OTEL_TRACE_PORT", "4318")
+
 
 def configure_tracer(
     *,
     enabled: bool,
-    endpoint: str,
     service_name: str,
+    exporter: Exporter = Exporter(
+        endpoint=f"http://{_env_host}:{_env_port}", exporter_type=ExporterType.HTTP
```
Comment (eirsyl) [inline troncos/tracing/__init__.py:20]: This is not correct? Add `/v1/trace`?

Comment (gbbirkisson) [reply troncos/tracing/__init__.py]: You are right, fixed ...

File: README.md (hunk 1 above)
Comment (gbbirkisson) [reply README.md:88]: Good idea, I simplified it even more so you only need to specify host in most cases.

File: troncos/tracing/_exporter.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        assert path, "You have to specify 'path'"
+
+        if not exporter_type:
+            if port == "4317":
+                exporter_type = ExporterType.GRPC
+            elif port == "4318":
+                exporter_type = ExporterType.HTTP
+        assert exporter_type, "You have to specify 'exporter_type'"
+
+        self.endpoint = f"{scheme}://{host}:{port}{path}"
```
Comment (eirsyl) [inline troncos/tracing/_exporter.py:36]: Make sure paths starts with `/`?

**Review by eirsyl — APPROVED** [review]

**Pull request merged**